duo-256.nix

```diff
@@ -190,17 +193,70 @@ in
     };
   };
 
-  users.users.root.initialHashedPassword = "";
+  users.users.root.initialPassword = "milkv";
   services.getty.autologinUser = "root";
 
   services.udev.enable = false;
   services.nscd.enable = false;
-  networking.firewall.enable = false;
-  networking.useDHCP = false;
   nix.enable = false;
   system.nssModules = lib.mkForce [ ];
 
-  environment.systemPackages = with pkgs; [ pfetch python311 ];
+  networking = {
+    interfaces.usb0 = {
+      ipv4.addresses = [
+        {
+          address = "192.168.42.1";
+          prefixLength = 24;
+        }
+      ];
+    };
+    useDHCP = false;
+    hostName = "nixos-duo";
+    firewall.enable = false;
+  };
+
+  systemd.services.rndis-device = {
+    enable = true;
+    description = "Set up usb0 as an RNDIS device";
+    unitConfig = {
+      Type = "oneshot";
+    };
+    serviceConfig = {
+      ExecStart = ''
+        /bin/sh -c "${pkgs.coreutils}/bin/echo device > /proc/cviusb/otg_role"
```

Comment:
Can we use `tmpfiles.d` for this.

Reply:
Do you have a link to a Nix example that is similar?

Reply:
```nix
{
  systemd.tmpfiles.settings = {
    "10-cviusb" = {
      "/proc/cviusb/otg_role".w.argument = "device";
    };
  };
}
```
Still there could be timing issues.

Reply:
That totally worked!  Way better, thanks.